src/App.js

```diff
@@ -30,17 +30,31 @@ const collectionName = "cs124-lab3";
 function App() {
     const [editingTaskId, setEditingTaskId] = useState("");
     const [showAlert, setShowAlert] = useState(false);
-    const [showPriorityDropdown, setShowPriorityDropdown] = useState("");
     const [showSortDropdown, setShowSortDropdown] = useState(false);
     const [tabIndex, setTabIndex] = useState(1);
     const [taskToDeleteParams, setTaskToDeleteParams] = useState(["", false]);
     const [sortDirection, setSortDirection] = useState("asc");
     const [sortParam, setSortParam] = useState("text");
     const q = query(collection(db, collectionName), orderBy(sortParam, sortDirection));
-    const [taskList, loading] = useCollectionData(q);
+    const [qSubParams, setQSubParams] = useState([db, collectionName, "", ""]);
```

Comment:
what about: useState(collection(db, collectionName, "", ""))
That way, you have only one object, instead of 4 to deal with.